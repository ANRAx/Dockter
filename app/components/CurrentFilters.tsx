import React from 'react';

const CurrentFilters = (props) => {
  const buttons = [];

  //iterates over filterOption state
  Object.keys(props.filterOptions).forEach((el) => {
    const option = props.filterOptions[el];

    //at each key check is property is an array
    if (Array.isArray(option) && option.length) {
      //if property is an array, make a button for each selection
      option.forEach((selection, i) => {
        buttons.push(
          <button
            key={i}
            onClick={() => {
              //if user clicks on button, it will remove the filter from filterOptions
              props.setFilterOptions({
                ...props.filterOptions,
                [el]: option.reduce((acc, element) => {
                  if (element !== selection) {
                    acc.push(element);
                  }
                  return acc;
                }, []),
              });
            }}
          >
            {selection}
          </button>
        );
      });
    }
  });

  return <div id="current-filters-container">{buttons}</div>;
};

export default CurrentFilters;
