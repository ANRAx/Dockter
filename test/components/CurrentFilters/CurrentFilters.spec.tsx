import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import CurrentFilters from '../../../app/components/CurrentFilters';

describe('CurrentFilters component', () => {
  const filterOptions = {
    container_id: [],
    container_name: [],
    container_image: [],
    status: [],
    stream: [],
    timestamp: {
      from: '',
      to: '',
    },
    host_ip: [],
    host_port: [],
    log_level: [],
  };

  it('should match exact snapshot', () => {
    //   // TODO: Implement hook test
    // const tree = renderer
    //   .create(<CurrentFilters filterOptions={filterOptions} />)
    //   .toJSON();
    // expect(tree).toMatchSnapshot();
  });
});
