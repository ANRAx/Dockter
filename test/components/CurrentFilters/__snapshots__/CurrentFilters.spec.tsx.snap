// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentFilters component should match exact snapshot 1`] = `
<div
  className="pt-5"
  id="current-filters-container"
/>
`;
